lms/djangoapps/ci_program/views.py

```diff
@@ -17,7 +18,10 @@ def show_programs(request, program_name):
     student_email = request.user.email
     cache_key = '%s_program_name' % student_email
 
-    program = Program.objects.get(marketing_slug=program_name)
+    try:
+        program = Program.objects.get(marketing_slug=program_name)
+    except Program.DoesNotExist:
+        raise Http404
```

Comment:
Was it not to redirect to the dashboard rather than raise a 404 error? You can leave it, but is the 404 page looking ok?

Reply:
The story does say to redirect to the dashboard but it seemed it could cause frustration. The 404 is basic, but has a link to the dashboard. It just means the user gets a bit of feedback about what they tried to do.

Do you think I should update it?

Reply:
No, that makes sense. Leave it as is so :+1: